index.js

```diff
@@ -6,11 +6,20 @@ const { GitHub, context } = require('@actions/github')
 const main = async () => {
   const token = core.getInput('github-token')
   const number = core.getInput('number')
+  const repo = core.getInput('repo')
+
+  var repoValue
+  if (!repo) {
+    repoValue = context.repo
+  } else {
+    const [owner, repo] = repo.split('/')
+    repoValue = {owner, repo}
+  }
 
   const octokit = new GitHub(token)
 
   await octokit.pulls.createReview({
-    ...context.repo,
+    ...repoValue,
```

Comment:
Hey, out of respect for your time and since I saw a few areas to improve, let me suggest an implementation here:

```js
  const repoString = core.getInput('repo')

  let repoObject
  if (repoString) {
    const [owner, repo] = repoString.split('/')
    repoObject = { owner, repo }
  } else {
    repoObject = context.repo
  }

  const octokit = new GitHub(token)

  await octokit.pulls.createReview({
    ...repoObject,
```

Main changes:

- use `let` instead of `var`, which is more explicit
- get around the awkward repo naming (it's not our fault "repo" can mean repository name or "owner/repo" string) by being explicit about it
- conditionals read easier if the positive condition comes first

Reply:
Thanks, I don't program in JavaScript except for very occasional contributions like this. 

I had to read about let vs var.   `let` is more scoped than `var` but just as explicit.  It is scoped to the exact block not just the function you are in.  scoping is good for many reasons.

Reply:
if you're interested, `let` is more explicit than `var` in that you expect the value of `let` to change, while if you have an unchangable value you use `const`. `var` is used for both purposes